src/workclaimer.js

```diff
@@ -0,0 +1,140 @@
+let assert  = require('assert');
+let _       = require('lodash');
+let Promise = require('promise');
+let events  = require('events');
+
+/**
+ * HintPoller polls for hints for pending tasks.
+ *
+ * The azure queues don't know if a task is pending they just store hints of
+ * pending tasks. To be understood this way:
+ *  A) If a task is pending, there is a hint of the task in an azure queue,
+ *  B) If there is an hint in an azure queue, it may or may not be pending.
+ *
+ * It's an if, but not an only-if (think over-approximation).
+ */
+class HintPoller {
+  constructor(parent, key) {
+    this.parent = parent;
+    this.key = key;
+    this.requests = [];
+    this.started = false;
+  }
+
+  requestClaim(count, aborted) {
+    // Make a request for count tasks
+    let request = null;
+    let result = new Promise(resolve => request = {resolve, count});
+    this.requests.push(request);
+
+    aborted.then(() => {
+      // Remove request from requests, and resolve request empty array
+      _.remove(this.requests, request);
+      request.resolve([]);
+    });
+
+    if (!this.started) {
+      this.start();
+    }
+
+    return result;
+  }
+
+  start() {
+    this.started = true;
+    this.poll().catch(err => this.parent.emit('error', err));
+  }
+
+  async poll() {
+    let count = 0;
+    while ((count = _.sumBy(this.requests, 'count')) > 0) {
+
+      let queues = this.owner.queueService.pendingQueues(
+        provisionerId, workerType,
+      );
+      for (let queue of queues) {
+        let hints = [];
+        while (count > 0 && (hints = await queue.poll(count)).length > 0) {
+          count -= hints.length;
+          this.claimHint
```

Comment:
are you missing something here?  I feel like there's a missing argument list to a function and a missing implementation